src/modules/party/party.controller.test.ts

```diff
@@ -0,0 +1,50 @@
+import { PartyGenerator } from '@ukef-test/support/generator/party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { when } from 'jest-when';
+
+import { PartyController } from './party.controller';
+import { PartyService } from './party.service';
+
+jest.mock('./party.service');
+
+describe('PartyController', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const partyIdentifier = valueGenerator.stringOfNumericCharacters();
+
+  let partyService: PartyService;
+  let controller: PartyController;
+
+  beforeEach(() => {
+    partyService = new PartyService(null, null);
+    controller = new PartyController(partyService);
+  });
+
+  describe('getPartyByIdentifier', () => {
+    const { parties, partiesFromApi } = new PartyGenerator(valueGenerator).generate({ numberToGenerate: 1 });
+    const partyFromService = parties[0];
+    const expectedParty = partiesFromApi[0];
+
+    it('returns the party from the service', async () => {
+      // eslint-disable-next-line jest/unbound-method
+      when(partyService.getPartyByIdentifier).calledWith(partyIdentifier).mockResolvedValueOnce(partyFromService);
+
+      const party = await controller.getPartyByIdentifier(partyIdentifier);
+
+      expect(party).toStrictEqual(expectedParty);
```

Comment:
Can we please add `status` and `data` property check for a more accurate assertion.

Reply:
We don't have `status` or `data` fields at this point: as it's a unit test, we're only executing the `controller.getPartyByIdentifier` method as a normal function. It doesn't have the NestJs framework around it to convert the result into a HTTP response with a status code

The API tests _do_ cover checking the status and data fields though